hoot-core/src/main/cpp/hoot/core/ops/CalculateStatsOp.cpp

```diff
@@ -144,178 +144,97 @@ void CalculateStatsOp::apply(const OsmMapPtr& map)
 
   MapProjector::projectToPlanar(map);
 
-  std::shared_ptr<const OsmMap> constMap = map;
-
-  _stats.append(SingleStat("Nodes",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Node),
-      ConstElementVisitorPtr(new ElementCountVisitor())))));
-  _stats.append(SingleStat("Ways",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Way),
-      ConstElementVisitorPtr(new ElementCountVisitor())))));
-  _stats.append(SingleStat("Relations",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Relation),
-      ConstElementVisitorPtr(new ElementCountVisitor())))));
-  _stats.append(SingleStat("Minimum Node ID",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Node),
-      ConstElementVisitorPtr(new MinIdVisitor())))));
-  _stats.append(SingleStat("Maximum Node ID",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Node),
-      ConstElementVisitorPtr(new MaxIdVisitor())))));
-  _stats.append(SingleStat("Minimum Way ID",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Way),
-      ConstElementVisitorPtr(new MinIdVisitor())))));
-  _stats.append(SingleStat("Maximum Way ID",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Way),
-      ConstElementVisitorPtr(new MaxIdVisitor())))));
-  _stats.append(SingleStat("Minimum Relation ID",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Relation),
-      ConstElementVisitorPtr(new MinIdVisitor())))));
-  _stats.append(SingleStat("Maximum Relation ID",
-    _applyVisitor(constMap, FilteredVisitor(ElementTypeCriterion(ElementType::Relation),
-      ConstElementVisitorPtr(new MaxIdVisitor())))));
+  _constMap = map;
 
-  if (!_quick)
+  // These arrays will move to json files
+  StatData quickStats[] =
```

Comment:
@bwitham I was hoping to put way more of these stats into the data array. But I pretty quickly hit the wall with all the special code and checks needed for the stats that are currently not in these arrays. 
I'm not sure how useful this is and if I should even consider putting this in a json config file when there is so much for which this won't work. I guess it would need a js script to make this fully data driven.
Do you have any suggestions? Any idea of a better way to abstract this?

Reply:
Ok, thanks. I didn't think this would be easy. I'll take a look tomorrow morning and see if I have any ideas.